### PR TITLE
Remove deprecated v4-6 REST API (`dspace-rest` module, also known as the `/rest` webapp)

### LINKED ISSUE #7643: [DS-4303] Old (v4-6) REST API is deprecated in v7, remove in v8
<i>Imported from JIRA <a href="https://jira.lyrasis.org/browse/DS-4303">[DS-4303]</a> created by tdonohue</i>
<p>As decided in DSpace 7 development, the DSpace v4-6 REST API (i.e. the old REST API or "rest" webapp in versions 4.x-6.x) is deprecated in version 7.x.  All tools should move to using the new REST API ("server" webapp in version 7.x)</p>

<p>This ticket is a reminder that this old REST API will be removed from the 8.0 release.  This ticket is logged as a "blocker" as it will need to be resolved prior to the 8.0 release.</p>

### PR DESCRIPTION
**Pull request by tdonohue**
## References

* Fixes #7643 

## Description
Removes the entire `dspace-rest` module (deprecated REST API from DSpace v4-6) along with all references to it.

This also includes the removal of a few `@Deprecated` methods **only used by `dspace-rest`**:
* `ItemService` and `ItemDAO` - `findByMetadataQuery()` method, deprecated in v7 via #2111 
    * This method was only kept to support `dspace-rest` as it uses deprecated features in Hibernate.  It must be removed to eventually support upgrading to Hibernate 6 (as the features used have been removed in Hiberate 6)
* `org.dspace.servicemanager.servlet.*` classes in `dspace-services`.  These supported the old way of starting the DSpace Kernel (via `DSpaceKernelServletContextListener`) which was only necessary in DSpace 6 and `dspace-rest`.  Removed as it's now unused/obsolete.

## Instructions for Reviewers

* No new code is added. 
* Simply verify that the DSpace backend still builds & installs properly without `dspace-rest`.

### CODE REVIEW

File: dspace/config/modules/rest.cfg [hunk 1]
```diff
@@ -54,143 +54,4 @@ rest.properties.exposed = google.recaptcha.mode
...
-
-# regex to identify partial XML entities within a description field (a frequent problem found in ProQuest ETD's)
-rest.report-regex-xml-entity = ^.*&#.*$
-
-# regex to identify non ascii characters in metadata
-rest.report-regex-non-ascii = ^.*[^\\p{ASCII}].*$
-
-# The maximum number of results to return for 1 request
-rest.search.max.results = 100
+rest.properties.exposed = handle.canonical.prefix
```
Comment (rmathew1011) [inline dspace/config/modules/rest.cfg:57]: New line

**Review by rmathew1011 — CHANGES_REQUESTED** [review]
new line for rest.cfg file.

File: dspace/config/modules/rest.cfg (hunk 1 above)
Comment (tdonohue) [reply dspace/config/modules/rest.cfg]: Thanks @rmathew1011 for the review.  To be honest, I don't think having a blank line at the end of these cfg files makes a difference :)  But I went ahead and added it back anyways.  So this is resolved.  (Note: I rebased this tiny change into the existing commit so there is no separate commit)

**Comment by hardyoyo** [comment]
tested locally with Docker-Compose, this builds and starts correctly. No errors observed in the logs. No further testing required. 👍

**Review by hardyoyo — APPROVED** [review]
👍

**Comment by tdonohue** [comment]
Merging with one approval. This is necessary to allow us to upgrade to Hibernate 6. We MUST upgrade to Hibernate 6 to fix #8713 (Tomcat 10 support requires latest Hibernate & Spring).

**Pull request merged**